web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -180,11 +180,20 @@
   }
 
   onMount(async () => {
-    unsubscribe = websocketEvents.on('on_upload_success', (assetUpdate) => {
-      if (assetUpdate.id === asset.id) {
-        asset = assetUpdate;
-      }
-    });
+    unsubscribes.push(
+      websocketEvents.on('on_upload_success', (assetUpdate) => {
+        if (assetUpdate.id === asset.id) {
+          asset = assetUpdate;
+        }
+      }),
+      websocketEvents.on('on_asset_update', (assetUpdate) => {
+        console.log('assetUpdate', assetUpdate);
+        if (assetUpdate.id === asset.id) {
+          asset = assetUpdate;
+        }
```

Comment:
Might be worth extracting the event handler, since it is exactly the same for both.